### PR TITLE
Ontonotes v5 Converter

### PR DESCRIPTION
**Pull request by rahul1980**
The converter  handles all annotations in the Ontonotes v5 corpus, performs various span normalizations and writes the output as SLING documents, along with a detailed summary of the input, normalization, and output. 

Supported span normalization heuristics:
- Drop leading articles.
- Descend from prepositions to their objects.
- Shrinking a span to another if they share the same head.
- Particle inclusion in SRL predicates.

Each normalization can be turned on/off on the commandline.
Other commandline flags include:
- Importing coreference annotations or not (default=not)
- Output per sentence documents (default=yes)

This PR also extends some util scripts, and makes a few fixes after the symbol-shortening change.

### CODE REVIEW

File: sling/nlp/parser/tools/commons_from_corpora.py [hunk 1]
```diff
@@ -74,15 +74,19 @@ def add(handle):
                default="",
                type=str,
                metavar='COMMA_SEPARATED_RECORDIOS')
-
   flags.define('--output',
                help='Output commons file name',
                default="/tmp/commons.sling",
                type=str,
                metavar='OUTPUT_COMMONS_FILENAME')
+  flags.define('--binary',
```
Comment (ringgaard) [inline sling/nlp/parser/tools/commons_from_corpora.py]: I think we should have binary=True as the default.

File: python/nlp/document.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ class DocumentSchema:
   def __init__(self, store):
     self.isa = store['isa']
     self.document = store['document']
+    self.document_id = store['document_id']
```
[the remark this answers is not in the document]
Comment (rahul1980) [reply python/nlp/document.py]: Replaced with id.

File: python/nlp/document.py [hunk 3]
```diff
@@ -40,12 +41,21 @@ def __init__(self, store):
     self.token_start = store['start']
     self.token_size = store['size']
     self.token_break = store['break']
+    self.token_pos = store['pos']
```
[the remark this answers is not in the document]
Comment (rahul1980) [reply python/nlp/document.py]: Done

File: python/nlp/document.py [hunk 4]
```diff
@@ -40,12 +41,21 @@ def __init__(self, store):
...
 
     self.phrase = store['phrase']
     self.phrase_begin = store['begin']
     self.phrase_length = store['length']
     self.phrase_evokes = store['evokes']
 
+    self.thing = store['thing']
+    self.constituents = store['constituents']
```
[the remark this answers is not in the document]
Comment (rahul1980) [reply python/nlp/document.py]: Added constituency.sling in the schemas folder.

File: python/nlp/document.py [hunk 5]
```diff
@@ -262,8 +272,7 @@ def decorate(self):
       if t.frame.word == None: t.word = self.text[t.start:t.end]
       index += 1
     for m in self.mentions:
-      if m.frame.name == None:
-        m.frame.name = self.phrase(m.begin, m.end)
+      m.frame.name = self.phrase(m.begin, m.end)
```
[the remark this answers is not in the document]
Comment (rahul1980) [reply python/nlp/document.py]: I was getting an attribute not found error for some reason but rarely. Reverted.

File: sling/nlp/parser/ontonotes/ontonotesv5_to_sling.py [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Main script for converting .gold_conll files from Ontonotes v5 to SLING.
+
+Sample usage:
+ONTO=path/to/ontonotes/data/train/data/english/annotations/
+python path/to/this/script.py --output=/tmp/output.rio --input=$ONTO
```
[the remark this answers is not in the document]
Comment (rahul1980) [reply sling/nlp/parser/ontonotes/ontonotesv5_to_sling.py]: Fixed.

File: sling/nlp/parser/tools/commons_from_corpora.py (hunk 1 above)
Comment (rahul1980) [reply sling/nlp/parser/tools/commons_from_corpora.py]: Done.

**Pull request merged**